msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AcquireTokenSilentlyTest.java

```diff
@@ -45,7 +56,71 @@ void confidentialAppAcquireTokenSilently_emptyCache_MsalClientException() throws
 
         ExecutionException ex = assertThrows(ExecutionException.class, future::get);
 
-        assertTrue(ex.getCause() instanceof MsalClientException);
+        assertInstanceOf(MsalClientException.class, ex.getCause());
         assertTrue(ex.getMessage().contains(AuthenticationErrorMessage.NO_TOKEN_IN_CACHE));
     }
+
+    @Test
+    void publicAppAcquireTokenSilently_claimsSkipCache() throws Throwable {
+
+        PublicClientApplication application = PublicClientApplication.builder("client_id")
+                .instanceDiscovery(false)
+                .authority("https://some.authority.com/realm")
+                .build();
+
+        application.tokenCache.deserialize(cache);
+
+        SilentParameters parameters = SilentParameters.builder(Collections.singleton("scopes"), basicAccount).build();
+
+        IAuthenticationResult result = application.acquireTokenSilently(parameters).get();
+
+        //Confirm cached dummy token returned from silent request
+        assertNotNull(result);
+        assertEquals("token", result.accessToken());
+
+        ClaimsRequest cr = new ClaimsRequest();
```

Comment:
This is not good API experience ( I know it exists already).

 Is there an overload where the claims are injected as a string? After all, this is what the app developers will use.

Reply:
For simplicity I'm not deprecating in this PR/release, but I created this task to get it deprecated in the next one: https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/855